echidna/core/fill_spectrum.py

```diff
@@ -8,427 +8,296 @@
-    """**Weights have been disabled.**
-    This function fills in the ndarray of energies, radii, times and
-    weights. It takes the reconstructed energies and true positions of
-    the events from the ntuple. In order to keep the statistics, the
-    time dependence is performed via adding weights to every event
-    depending on the time period. Both, studied time and Half-life must
-    be written in the same units.
-
-    Args:
-      filename (str): The ntuple to study
-      T (float): The Half-life of a studied background
-      spectrumname (str, optional): A name of future spectrum. Not
-        required when appending a spectrum.
-      spectrum (:class:`echidna.core.spectra.Spectra`, optional):
-        Spectrum you wish to append. Not required when creating a
-        new spectrum.
-
-    Raises:
-      ValueError: If spectrumname is not set when creating a new
-        spectrum.
-
-    Returns:
-      spectrum (:class:`echidna.core.spectra.Spectra`)
-    """
-    print filename
-    chain = TChain("output")
-    chain.Add(filename)
-    if spectrum is None:
-        if spectrumname == "":
-            raise ValueError("Name not set when creating new spectra.")
-        spectrum = spectra.Spectra(str(spectrumname), 10.*chain.GetEntries())
-    else:
-        spectrum._num_decays += 10.*chain.GetEntries()
-        spectrumname = spectrum._name
-    print spectrumname
-
-    times = []
-    for time_step in range(0, spectrum._time_bins):
-        time = time_step * spectrum._time_width + spectrum._time_low
-        times.append(time)
-
-    if 'AV' in spectrumname:
-        print "AV WEIGHTS ARE CURRENTLY UNAVAILABLE"
-        weights = _av_weights(times, T)
-    else:
-        weights = _scint_weights(times, T)
-
-    for event in chain:
-        if event.scintFit == 0:
+        fill = True
+        fill_kwargs = {}
+        if bipo and ds.GetEVCount() != 1:
+            # Only bipos with 1 ev survive bipo cut
             continue
-        energy = event.energy
-        position = math.fabs(math.sqrt((event.posx)**2 +
-                                       (event.posy)**2 + (event.posz)**2))
-        spectrum._raw_events += 1
-        for time, weight in zip(times, weights):
+        for var, extractor in extractors.iteritems():
+            if extractor.get_valid_root(mc):
+                fill_kwargs[extractor._name] = extractor.get_value_root(mc)
+            else:
+                fill = False
+                break
+        if fill:
             try:
-                spectrum.fill(energy, position, time, 1.)
+                spectrum.fill(**fill_kwargs)
+                spectrum._raw_events += 1
             except ValueError:
                 pass
 
-    return spectrum
-
 
-def fill_mc_ntuple_spectrum(filename, T, spectrumname="", spectrum=None):
+def fill_from_root(filename, spectrum_name="", config=None, spectrum=None,
+                   bipo=False, **kwargs):
     """**Weights have been disabled.**
```

Comment:
I think we can remove these lines now. There are also some references to weights in the lines below, which I'm not entirely sure about.